src/components/ciConfig/CIConfigDiffView.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { useEffect } from 'react'
+import { noop, Progressing, VisibleModal } from '../common'
+import { ReactComponent as CloseIcon } from '../../assets/icons/ic-cross.svg'
+import { Workflow } from '../workflowEditor/Workflow'
+import { Link, useHistory, useLocation, useParams, useRouteMatch } from 'react-router-dom'
+import { URLS } from '../../config'
+
+export default function CIConfigDiffView({
+    ciConfig,
+    configOverridenPipelines,
+    configOverrides,
+    processedWorkflows,
+    toggleConfigOverrideDiffModal,
+}) {
+    const history = useHistory()
+    const location = useLocation()
+    const match = useRouteMatch<{
+        appId: string
+    }>()
+    const { appId } = useParams<{
+        appId: string
+    }>()
+    const wfCIMap = new Map<number, number>()
+    const _configOverridenWorkflows = configOverrides?.workflows?.filter((_cwf) => {
+        const _ciPipeline = configOverridenPipelines.find((_ci) => _ci.id === _cwf.ciPipelineId)
+        if (!!_ciPipeline) {
+            wfCIMap.set(_cwf.id, _ciPipeline.id)
+            return _ciPipeline
+        }
+    })
+    const _overridenWorkflows = processedWorkflows.workflows.filter(
+        (_wf) => !!_configOverridenWorkflows.find((_cwf) => _cwf.id === +_wf.id),
+    )
+    const globalCIConfig = {
+        dockerRegistry: ciConfig?.dockerRegistry,
+        dockerRepository: ciConfig?.dockerRepository,
+        dockerBuildConfig: {
+            gitMaterialId: ciConfig?.dockerBuildConfig?.gitMaterialId,
+            dockerfileRelativePath: ciConfig?.dockerBuildConfig?.dockerfileRelativePath,
+        },
+    }
+
+    const renderDetailedValue = (parentClassName: string, title: string, value: string) => {
+        return (
+            <div className={parentClassName}>
+                <div className="cn-6 pt-8 pl-16 pr-16 lh-16">{title}</div>
+                <div className="cn-9 fs-13 pb-8 pl-16 pr-16 lh-20 mh-28">{value}</div>
+            </div>
+        )
+    }
+
+    const renderValueDiff = (baseValue, currentValue, changedBGColor, configName) => {
+        return (
+            <>
+                {baseValue ? (
+                    renderDetailedValue(changedBGColor ? 'code-editor-red-diff' : '', configName, baseValue)
+                ) : (
+                    <div />
+                )}
+                {currentValue ? (
+                    renderDetailedValue(changedBGColor ? 'code-editor-green-diff' : '', configName, currentValue)
+                ) : (
```

Comment:
why div and not null?

Reply:
As discussed, an empty div is required for supporting the grid layout that we're trying to create.